File: Leetcode/448-Find-All-Numbers-Disappeared-in-an-Array/448-Find-All-Numbers-Disappeared-in-an-Array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        arr = set(nums)
        result = []
        n = len(nums)
        for i in range(1, n + 1):
            if i not in arr:
                result.append(i)
        return result

File: Leetcode/448-Find-All-Numbers-Disappeared-in-an-Array/448-Find-All-Numbers-Disappeared-in-an-Array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        Set<Integer> missing = new TreeSet<>();

        for (int i = 1; i <= nums.length; i++) {
            missing.add(i);
        }

        for (int num : nums) {
            missing.remove(num);
        }

        List<Integer> result = new ArrayList<>(missing);

        return result;
    }
}